debugger/src/debugger/debug_entry.rs

```diff
@@ -208,81 +211,91 @@ impl Debugger {
+                                Ok(DebugSessionStatus::Continue)
+                            }
+                        }
                         "next" => debug_adapter
                             .next(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "stepIn" => debug_adapter
                             .step_in(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "stepOut" => debug_adapter
                             .step_out(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "pause" => debug_adapter
                             .pause(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "readMemory" => debug_adapter
                             .read_memory(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "writeMemory" => debug_adapter
                             .write_memory(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "setVariable" => debug_adapter
                             .set_variable(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "configurationDone" => debug_adapter
                             .configuration_done(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "threads" => debug_adapter
                             .threads(&mut target_core, request)
-                            .and(Ok(DebuggerStatus::ContinueSession)),
+                            .and(Ok(DebugSessionStatus::Continue)),
                         "restart" => {
                             // Reset RTT so that the link can be re-established
                             target_core.core_data.rtt_connection = None;
-                            debug_adapter
-                                .restart(&mut target_core, Some(request))
-                                .and(Ok(DebuggerStatus::ContinueSession))
+                            target_core
```

Comment:
I have had issues before where the reset timed out because of core activity, and since then, I default to making sure it is halted before I issue a reset. Is there a specific concern with forcing the halt a this point, or is this just an observation?